statistic/src/main/java/ch/unibas/dmi/dbis/polyphenydb/statistic/StatisticQueryProcessor.java

```diff
@@ -0,0 +1,452 @@
+     */
+    public StatisticQueryProcessor( final TransactionManager transactionManager, String userName, String databaseName ) {
+        this.transactionManager = transactionManager;
+        this.databaseName = databaseName;
+        this.userName = userName;
+    }
+
+
+    public StatisticQueryProcessor( TransactionManager transactionManager, Authenticator authenticator ) {
+        this( transactionManager, "pa", "APP" );
+    }
+
+
+    /**
+     * Handles the request for one columns stats
+     *
+     * @param query the select query
+     * @return result of the query
+     */
+    public StatisticQueryColumn selectOneStat( String query ) {
+        StatisticResult res = this.executeSqlSelect( query );
+        if ( res.getColumns() != null && res.getColumns().length == 1 ) {
+            return res.getColumns()[0];
+        }
+        return null;
+    }
+
+
+    /**
+     * Handles the request which retrieves the stats for multiple columns
+     */
+    public StatisticResult selectMultipleStats( String query ) {
+        return this.executeSqlSelect( query );
+    }
+
+
+    /**
+     * Method checks is limit is the same as buffer for statistics
+     * adds/matches is to the value in the configManager
+     *
+     * @param query the query to check
+     * @return the "sanitized" query
+     */
+    private String queryLimitCheck( String query ) {
+        Pattern pattern = Pattern.compile( "^.*(?=(limit))" );
+        return pattern.matcher( query ) + " LIMIT " + ConfigManager.getInstance().getConfig( "StatisticSize" ) + ";";
+    }
+
+
+    /**
+     * Method to get all schemas, tables, and their columns in a database
+     */
+    public List<List<String>> getSchemaTree() {
+
+        List<List<String>> result = new ArrayList<>();
+
+        Transaction transaction = getTransaction();
+        try {
+
+            CatalogDatabase catalogDatabase = transaction.getCatalog().getDatabase( databaseName );
+            CatalogCombinedDatabase combinedDatabase = transaction.getCatalog().getCombinedDatabase( catalogDatabase.id );
+            List<String> schemaTree = new ArrayList<>();
+            for ( CatalogCombinedSchema combinedSchema : combinedDatabase.getSchemas() ) {
+                // schema
+                // schemaTree.add( combinedSchema.getSchema().name );
+
+                List<String> tables = new ArrayList<>();
+
+                for ( CatalogCombinedTable combinedTable : combinedSchema.getTables() ) {
+
+                    List<String> table = new ArrayList<>();
+
+                    for ( CatalogColumn catalogColumn : combinedTable.getColumns() ) {
+                        table.add( combinedSchema.getSchema().name + "." + combinedTable.getTable().name + "." + catalogColumn.name );
+                    }
+
+                    if ( combinedTable.getTable().tableType.equals( "TABLE" ) ) {
+                        tables.addAll( table );
+                    }
```

Comment:
I'm not sure if this is working. You are comparing an Enum with a string. Correct would be
```java
if ( combinedTable.getTable().tableType == TableType.TABLE ) {
    tables.addAll( table );
}
```